conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/PlainSerDe.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.palantir.conjure.java.lib.SafeLong;
+import com.palantir.ri.ResourceIdentifier;
+import com.palantir.tokens.auth.BearerToken;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+/**
+ * Provides functionality to parse supported types using the
+ * <a href="https://github.com/palantir/conjure/blob/master/docs/spec/wire.md#6-plain-format">Conjure PLAIN format</a>.
+ *
+ * These utilities are used to parse HTTP path, query, and header parameter values.
+ */
+public interface PlainSerDe {
+
+    // TODO(ckozak): Write javadoc
+
+    BearerToken deserializeBearerToken(String in);
+
+    BearerToken deserializeBearerToken(Iterable<String> in);
```

Comment:
I don't really understand the signature of this... seems weird that we're asking implementers of this to make decisions about which of the iterable of strings to pick?  Why not just assert there's only one and pass to the `BearerToken  deserializeBearerToken(String)` method?

Reply:
That is logic we would have to implement in generated code, it’s cleaner to move that burden to code we write in the runtime and reuse.

Re nullable: that’s probably a bug, I will verify

Reply:
Fixed